perun-core/src/test/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntryIntegrationTest.java

```diff
@@ -2696,7 +2696,33 @@ public void autoAssignSubgroupInactive() throws Exception {
 		List<AssignedGroup> assignedGroups = sess.getPerun().getResourcesManager().getGroupAssignments(sess, inactiveResource, List.of());
 
 		AssignedGroup assignedGroup = new AssignedGroup(new EnrichedGroup(group, List.of()), GroupResourceStatus.INACTIVE, null, null, true);
-		AssignedGroup assignedSubgroup = new AssignedGroup(new EnrichedGroup(subGroup, List.of()), GroupResourceStatus.INACTIVE, group.getId(), null, true);
+
+		// automatically assigned subgroups should always be ACTIVE
+		AssignedGroup assignedSubgroup = new AssignedGroup(new EnrichedGroup(subGroup, List.of()), GroupResourceStatus.ACTIVE, group.getId(), null, true);
+
+		assertThat(assignedGroups).containsExactlyInAnyOrder(assignedGroup, assignedSubgroup);
+	}
+
+	@Test
+	public void assignGroupsToResourceWithInactiveSourceGroupWithSubgroup() throws Exception {
+		System.out.println(CLASS_NAME + "autoAssignSubgroupsWithInactiveSourceGroup");
+
+		vo = setUpVo();
+		member = setUpMember(vo);
+		facility = setUpFacility();
+		Resource inactiveResource = setUpResource();
+
+		group = setUpGroup(vo, member);
+		subGroup = setUpSubGroup(group);
+
+		sess.getPerun().getResourcesManager().assignGroupToResource(sess, group, inactiveResource, false, true, true);
+
+		List<AssignedGroup> assignedGroups = sess.getPerun().getResourcesManager().getGroupAssignments(sess, inactiveResource, List.of());
+
+		AssignedGroup assignedGroup = new AssignedGroup(new EnrichedGroup(group, List.of()), GroupResourceStatus.INACTIVE, null, null, true);
+
+		// automatically assigned subgroups should always be ACTIVE
```

Comment:
Now, we won't need this comment if we know what the code should do thanks to the test name. In general, it is good to avoid inline comments if possible, as code should be as self-explanatory as possible and inline comments are then often useless.